metadata-ingestion/src/datahub/ingestion/source/sql/presto_on_hive.py

```diff
@@ -166,7 +205,7 @@ class PrestoOnHiveSource(SQLAlchemySource):
     """
 
     def __init__(self, config: PrestoOnHiveConfig, ctx: PipelineContext) -> None:
-        super().__init__(config, ctx, "presto-on-hive")
+        super().__init__(config, ctx, config.mode)
```

Comment:
`s/config.mode/config.mode.value/` ?

Reply:
config.mode is a string. I think pydantic converts it. 

Tried it: 
```
from pydantic import BaseModel, Field
from typing import Optional

class TestConfig(BaseModel):
    b: Optional[str]

conf =  TestConfig(b='x')
conf.b
```
This returns with 'x'